de-lib/src/main/java/org/iplantc/de/client/services/converters/AppTemplateDECallbackConverter.java

```diff
@@ -0,0 +1,165 @@
+package org.iplantc.de.client.services.converters;
+
+import org.iplantc.de.client.models.apps.integration.AppTemplate;
+import org.iplantc.de.client.models.apps.integration.AppTemplateAutoBeanFactory;
+import org.iplantc.de.client.models.apps.integration.Argument;
+import org.iplantc.de.client.models.apps.integration.ArgumentGroup;
+import org.iplantc.de.client.models.apps.integration.ArgumentType;
+import org.iplantc.de.client.models.apps.integration.ArgumentValidator;
+import org.iplantc.de.client.models.apps.integration.SelectionItem;
+import org.iplantc.de.client.models.apps.integration.SelectionItemGroup;
+import org.iplantc.de.client.util.AppTemplateUtils;
+import org.iplantc.de.shared.DECallback;
+
+import com.google.common.collect.Lists;
+import com.google.web.bindery.autobean.shared.AutoBean;
+import com.google.web.bindery.autobean.shared.AutoBeanCodex;
+import com.google.web.bindery.autobean.shared.AutoBeanUtils;
+import com.google.web.bindery.autobean.shared.Splittable;
+import com.google.web.bindery.autobean.shared.impl.StringQuoter;
+
+import java.util.List;
+
+/**
+ * @author jstroot
+ */
+public class AppTemplateDECallbackConverter extends DECallbackConverter<String, AppTemplate> {
```

Comment:
I'm not finding that terrain `/preview-template` endpoint.
We have [POST /apps/arg-preview](https://cyverse-de.github.io/api/endpoints/app-metadata.html#previewing-command-line-arguments) that is an apps service endpoint.